src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -72,10 +73,79 @@ float64_t CGaussianARDKernel::compute(int32_t idx_a, int32_t idx_b)
 	return CMath::exp(-result);
 }
 
+SGVector<float64_t> CGaussianARDKernel::get_parameter_gradient_diagonal(
+		const TParameter* param, index_t index)
+{
+	REQUIRE(param, "Param not set\n");
+	REQUIRE(lhs , "Left features not set!\n");
+	REQUIRE(rhs, "Right features not set!\n");
+
+	if (lhs==rhs) 
```

Comment:
if the left features and right features are not the same object but do equal, we `cannot` use this trick.